clients/web/src/views/widgets/MetadataWidget.js

```diff
@@ -2,80 +2,418 @@
  * This widget shows a list of metadata in a given item.
  */
 girder.views.MetadataWidget = girder.View.extend({
+    modes: {
+        simple: {
+            editor: function (args) {
+                return new girder.views.MetadatumEditWidget(args);
+            },
+            displayValue: function () {
+                return this.value;
+            },
+            template: girder.templates.metadatumView
+        },
+        json: {
+            editor: function (args) {
+                if (args.value) {
+                    args.value = JSON.parse(args.value);
+                }
+                return new girder.views.JsonMetadatumEditWidget(args);
+            },
+            displayValue: function () {
+                return JSON.stringify(this.value, null, 4);
+            },
+            validation: {
+                from: {
+                    simple: [
+                        function (value) {
+                            try {
+                                var jsonValue = JSON.parse(value);
+                                /* This may succeed when we don't want it to (for instance with the
+                                 * value 'false' or '1234'), so check and only switch to JSON if we
+                                 * got an object back. */
+                                if (jsonValue && typeof jsonValue === 'object' && jsonValue !== null) {
```

Comment:
The third expression `jsonValue !== null` is redundant. Also, should we really forbid scalars like `1234` and `false` since they are also valid JSON? I feel like we should just return true if JSON.parse didn't throw an exception.